### PR TITLE
fix #1176 implemented GraticuleLayer for leaflet

### LINKED ISSUE #1176: Implement a Graticule Layer for Leaflet
Currently the GraticuleLayer is only supported on OL3 and CesiumJS

### PR DESCRIPTION
**Pull request by saidaipparla**
- added unit test too

### CODE REVIEW

File: web/client/components/map/leaflet/plugins/GraticuleLayer.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2015, GeoSolutions Sas.
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/GraticuleLayer.js]: 2016

File: web/client/components/map/leaflet/plugins/GraticuleLayer.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const SimpleGraticule = require('leaflet-simple-graticule/L.SimpleGraticule');
+const assign = require('object-assign');
+
+require('leaflet-simple-graticule/L.SimpleGraticule.css');
+
+Layers.registerType('graticule', {
+    create: (options) => {
+        const graticuleOptions = assign({
+            interval: 20,
+            showshowOriginLabel: true,
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/GraticuleLayer.js]: showOriginLabel

File: web/client/components/map/leaflet/plugins/GraticuleLayer.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            redraw: 'move'
+        }, options);
+        if (SimpleGraticule) {
+            return new SimpleGraticule(graticuleOptions);
+        }
+        return false;
```
Comment (mbarto) [inline web/client/components/map/leaflet/plugins/GraticuleLayer.js]: return null;

File: web/client/components/map/leaflet/plugins/GraticuleLayer.js (hunk 2 above)
Comment (saidaipparla) [inline web/client/components/map/leaflet/plugins/GraticuleLayer.js]: should i keep this showOriginLabel:(0,0) as initiial parameter ?

Comment (saidaipparla) [inline web/client/components/map/leaflet/plugins/GraticuleLayer.js]: ok

**Pull request merged**